addons/website_theme_install/models/ir_module_module.py

```diff
@@ -154,30 +185,23 @@ def _update_records(self, old, new):
                 else:
                     created += self.env[model].create(rec_data)
                 remaining -= rec
+        if len(remaining):
+            _logger.info('Error - Remaining: %s' % remaining.mapped('display_name'))
 
         deleted = old - updated
         deleted.unlink()
         return (created, updated, deleted)
 
-    def _load_one_theme_module(self, website, with_update=True, **kw):
+    def _load_one_theme_module(self, website, **kw):
         # load data from xml to template table
-        old_menus = self._get_module_data('theme.website.menu').with_context(active_test=False).mapped('copy_ids').filtered(lambda m: m.website_id == website)
-        old_pages = self._get_module_data('theme.website.page').with_context(active_test=False).mapped('copy_ids').filtered(lambda p: p.website_id == website)
-        old_attachs = self._get_module_data('theme.ir.attachment').with_context(active_test=False).mapped('copy_ids').filtered(lambda a: a.website_id == website)
-        old_views = self._get_module_data('theme.ir.ui.view').with_context(active_test=False).mapped('copy_ids').filtered(lambda v: v.website_id == website)
-
-        if with_update:
```

Comment:
Computing "old" after the upgrade might not work correctly or at all, need more testing.

Is there a correct place to plug some code before the module even starts to upgrade?